### PR TITLE
MYX Finance : DEX : tvl by user

### PR DESCRIPTION
**Pull request by marvin6en**
## Title: 

- [x] Pattern for PR title "protocol_name : protocol_category : other comments". Example - "uniswap: dex : tvl by user"

## Checklist before requesting a review
1. **index.ts file**

     - [x] **Contains function**

            ```export const getUserTVLByBlock = async (blocks: BlockData) => {
                const { blockNumber, blockTimestamp } = blocks
                    //    Retrieve data using block number and timestamp
                    // YOUR LOGIC HERE
                
                return csvRows

            };
            ``` 
    - [x] **getUserTVLByBlock function takes input with this schema**
        
            ``` 
                interface BlockData {
                    blockNumber: number;
                    blockTimestamp: number;
                }
            ```
    - [x] **getUserTVLByBlock function returns output in this schema**

                ```
                const csvRows: OutputDataSchemaRow[] = [];

                type OutputDataSchemaRow = {
                    block_number: number;  //block_number which was given as input
                    timestamp: number;     // block timestamp which was given an input, epoch format
                    user_address: string;   // wallet address, all lowercase
                    token_address: string;  // token address all lowercase
                    token_balance: bigint;  // token balance, raw amount. Please dont divide by decimals
                    token_symbol: string; //token symbol should be empty string if it is not available
                    usd_price: number; //assign 0 if not available
                };
                ```
    - [x] **contains function**

            ```
                const readBlocksFromCSV = async (filePath: string): Promise<BlockData[]> => {
                const blocks: BlockData[] = [];

                await new Promise<void>((resolve, reject) => {
                    fs.createReadStream(filePath)
                    .pipe(csv()) // Specify the separator as '\t' for TSV files
                    .on('data', (row) => {
                        const blockNumber = parseInt(row.number, 10);
                        const blockTimestamp = parseInt(row.timestamp, 10);
                        if (!isNaN(blockNumber) && blockTimestamp) {
                        blocks.push({ blockNumber: blockNumber, blockTimestamp });
                        }
                    })
                    .on('end', () => {
                        resolve();
                    })
                    .on('error', (err) => {
                        reject(err);
                    });
                });

                return blocks;
                };

            ```
    - [x] **has this code**
           
            ```
            readBlocksFromCSV('hourly_blocks.csv').then(async (blocks: any[]) => {
            console.log(blocks);
            const allCsvRows: any[] = []; 

            for (const block of blocks) {
                try {
                    const result = await getUserTVLByBlock(block);
                    allCsvRows.push(...result);
                } catch (error) {
                    console.error(`An error occurred for block ${block}:`, error);
                }
            }
            await new Promise((resolve, reject) => {
                const ws = fs.createWriteStream(`outputData.csv`, { flags: 'w' });
                write(allCsvRows, { headers: true })
                    .pipe(ws)
                    .on("finish", () => {
                    console.log(`CSV file has been written.`);
                    resolve;
                    });
            });

            }).catch((err) => {
            console.error('Error reading CSV file:', err);
            });
        ```
2. **Output data**
    - [x] Data is returned for underlying tokens only. Not for special tokens (lp/veTokens etc)
    - [x] Follows the exact sequence mentioned in OutputDataSchemaRow . This is needed as we want same column ordering in output csv
    - Value of each field is :
        - [x] block_number *is same as input block number. This signifies TVL is as of this block_number.*
        - [x] timestamp is same as input timestamp. This signifies TVL is as of this timestamp. It is in epoch format.
        - [x] user_address is in lowercase
        - [x] token_address is in lowercase
        - [x] token_balance is in raw amount. Please dont divide by decimals.
        - [x] token_symbol value if present, empty string if value is not available.
        - [x] usd_price if value is available, 0 if value is not available.

### CODE REVIEW

**Comment by marvin6en** [comment]
> @marvin6en saw the repo.
> 
> looks like currently the adapter is returning the lp address and lp balance?
> 
> we will need them to be on underlying token asset level :
> 
> eg : btc - usdc pair -> btc and usdc separately.
> 
> and sum up these underlying token balance, to get the total balance of each asset.
> 
> token balances will need bigINT (raw number)

@0xroll Already updated as requested✅

File: adapters/myx/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                if (chunk.block) blocks.push(parseInt(chunk.block, 10));
+            }
+        }
+    );
+    return blocks;
+};
+
+
+const getData = async () => {
+    const snapshotBlocks = [
```
Comment (nitish-91) [inline adapters/myx/src/index.ts:43]: blocks should be received as input parameter.

**Comment by nitish-91** [comment]
@marvin6en You have marked everything on the checklist but i dont see the code taking 
1. block data as input
2. having getUserTVLByBlock function.
and many more. Can you please look at the index file of existing protocols and include the differences

**Comment by marvin6en** [comment]
> @marvin6en You have marked everything on the checklist but i dont see the code taking
> 
> 1. block data as input
> 2. having getUserTVLByBlock function.
>    and many more. Can you please look at the index file of existing protocols and include the differences

@nitish-91 It has been fixed, please check again

File: adapters/myx/src/sdk/positionSnapshots.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        if (!userPositionSnapshotMap) {
+            userPositionSnapshotMap = new Map<string, LiquidityPositionSnapshot>()
+        }
+        userPositionSnapshotMap.set(snapshot.lPToken.address, snapshot)
+        snapshotsMap.set(snapshot.recipient, userPositionSnapshotMap)
+    }
+    snapshotsMap.forEach((userPositionSnapshotMap => {
+        userPositionSnapshotMap.forEach((positionSnapshot) => {
+            userPositionSnapshotsAtBlockData.push({
+                user_address: positionSnapshot.recipient,
```
Comment (nitish-91) [inline adapters/myx/src/sdk/positionSnapshots.ts:108]: Will this return multiple block numbers for 1 block input?

Same question for timestamp as well
@marvin6en

Comment (marvin6en) [reply adapters/myx/src/sdk/positionSnapshots.ts:108]: at line 55-58，the graphql Already filtered

**Pull request merged**